src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -60,45 +60,8 @@ def handleRequest(request)
         response["X-Puppet-Version"])
   end
 
-  def getClassInfoForAllEnvironments()
-    # The clear_environment_settings(env) ensures that the calls which happen
-    # later on to enumerate all of the manifests for each environment are
-    # using the latest environment.conf settings for each environment.  Without
-    # this call, cached (and, therefore, possibly stale) environment.conf
-    # settings would be used.  We want to return the latest data for any calls
-    # made to this method.
-    @env_loader.list.each do |env|
-      Puppet.settings.clear_environment_settings(env)
-    end
-
-    environments =
-      Hash[@env_loader.list.collect do |env|
-       [env.name, self.class.getManifests(env)]
-      end]
-
-    classes_per_env =
-        Puppet::InfoService::ClassInformationService.new.classes_per_environment(environments)
-    Hash[classes_per_env.collect {|key, value| [key.to_s, value]}]
-  end
-
   def getClassInfoForEnvironment(env)
-    # The clear_environment_settings(env) ensures that the calls which happen
-    # later on to enumerate all of the manifests for each environment are
-    # using the latest environment.conf settings for each environment.  Without
-    # this call, cached (and, therefore, possibly stale) environment.conf
-    # settings would be used.  We want to return the latest data for any calls
-    # made to this method.
-    Puppet.settings.clear_environment_settings(env)
```

Comment:
This call was in place to ensure that the latest settings from the environment's "environment.conf" file would be sourced before trying to parse class information.  This call is not necessary so long as the environment is marked as "expired" prior to this call being made.  As we now expect `mark-environments-expired!` calls to be part of the workflow for refreshing the cache, it seemed better to eliminate this call from the code - especially since it is an internal detail from Ruby Puppet which could change out from underneath Puppet Server Ruby code, potentially without warning.